admin/javascript/src/components/form-action/index.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import SilverStripeComponent from 'silverstripe-component';
+
+class FormActionComponent extends SilverStripeComponent {
+  constructor(props) {
+    super(props);
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  render() {
+    const props = {
+      type: this.props.type,
+      className: this.getButtonClasses(),
+      disabled: this.props.disabled,
+    };
+
+    if (typeof this.props.id !== 'undefined') {
```

Comment:
No, the spread operator will never delete keys. It's all driven by `hasOwnProperty`.